src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageService.MemoryMappedInfo.cs

```diff
@@ -22,47 +21,32 @@ internal partial class TemporaryStorageService
     /// metadata dll shadow copy. shared view will help those cases.
     /// </summary>
     /// <remarks>
-    /// <para>Instances of this class should be disposed when they are no longer needed. After disposing this
-    /// instance, it should no longer be used. However, streams obtained through <see cref="CreateReadableStream"/>
-    /// or <see cref="CreateWritableStream"/> will not be invalidated until they are disposed independently (which
-    /// may occur before or after the <see cref="MemoryMappedInfo"/> is disposed.</para>
-    ///
-    /// <para>This class and its nested types have familiar APIs and predictable behavior when used in other code,
-    /// but are non-trivial to work on. The implementations of <see cref="IDisposable"/> adhere to the best
-    /// practices described in
-    /// <see href="http://joeduffyblog.com/2005/04/08/dg-update-dispose-finalization-and-resource-management/">DG
-    /// Update: Dispose, Finalization, and Resource Management</see>. Additional notes regarding operating system
-    /// behavior leveraged for efficiency are given in comments.</para>
+    /// <para>This class and its nested types have familiar APIs and predictable behavior when used in other code, but
+    /// are non-trivial to work on.</para>
     /// </remarks>
-    internal sealed class MemoryMappedInfo(ReferenceCountedDisposable<MemoryMappedFile> memoryMappedFile, string name, long offset, long size) : IDisposable
+    internal sealed class MemoryMappedInfo(MemoryMappedFile memoryMappedFile, string name, long offset, long size)
     {
         /// <summary>
         /// The memory mapped file.
         /// </summary>
-        /// <remarks>
-        /// <para>It is possible for the file to be disposed prior to the view and/or the streams which use it.
-        /// However, the operating system does not actually close the views which are in use until the file handles
-        /// are closed as well, even if the file is disposed first.</para>
-        /// </remarks>
-        private readonly ReferenceCountedDisposable<MemoryMappedFile> _memoryMappedFile = memoryMappedFile;
```

Comment:
practically no one ever disposed this, so the udnerlying mmf was never disposed (i'll call out the one place we do it).  So this means the system has worked fine in al lthe places we use MMFs usnig normal gc+finalization logic for cleaniup.